src/TuDa.CIMS.Api/Repositories/AssetItemRepository.cs

```diff
@@ -0,0 +1,100 @@
+﻿using Microsoft.EntityFrameworkCore;
+using TuDa.CIMS.Api.Database;
+using TuDa.CIMS.Api.Interfaces;
+using TuDa.CIMS.Shared.Dtos;
+using TuDa.CIMS.Shared.Entities;
+
+namespace TuDa.CIMS.Api.Repositories;
+
+public class AssetItemRepository : IAssetItemRepository
+{
+    private readonly CIMSDbContext _context;
+
+    public AssetItemRepository(CIMSDbContext context)
+    {
+        _context = context;
+    }
+
+    /// <summary>
+    /// Returns all existing AssetItems of the database.
+    /// </summary>
+    public async Task<IEnumerable<AssetItem>> GetAllAsync()
+    {
+        return await _context.AssetItems.Include(i => i.Room).ToListAsync();
+    }
+
+    /// <summary>
+    /// Returns an existing AssetItem with the specific id.
+    /// </summary>
+    /// <param name="id">the unique id of the AssetItem</param>
+    public async Task<AssetItem?> GetOneAsync(Guid id)
+    {
+        return await _context.AssetItems.Where(i => i.Id == id).Include(i => i.Room).SingleAsync();
+    }
```

Comment:
🔧 Decide whether to handle errors with `Optionals` like in the return type or with Exceptions, done by the `SingleAsync`